examples/tts/conf/hifigan/hifigan_22050_data.yaml

```diff
@@ -0,0 +1,132 @@
+# This config contains the default values for training a 22khz HiFi-GAN model.
+# If you want to train model on other dataset, you can change config values according to your dataset.
+# Most dataset-specific arguments are in the head of the config file, see below.
+
+name: "HifiGan"
+
+max_epochs: ???
+batch_size: 16
+weighted_sample_steps: null
+
+train_ds_meta: ???
```

Comment:
Yes. I have added a usage example to the PR writeup.

Reply:
Do you think this these dataclasses could be turned into `TypedDict`?

The flow of "YAML dicts <-> Python dicts" is broken here, with additional verbosity.

Reply:
I have not used TypedDict much. My understanding is that the advantages/disadvantages of using it would be:

**Advantages**
1. Dict type is naturally supported by YAML config, avoiding repeated _target_ specification.
2. TypedDict is backwards compatible with code that previously used dictionary (not applicable here).

**Disadvantages**
1. Does not validate keys; user could be missing a required field or have a typo and the error would not be caught immediately.
2. Code using it is required to access fields like dictionary instead of object (ie. metadata["field"] vs metadata.field)

Of these I think the most important is dataclass validating that the correct keys are provided.

In the rest of NeMo, there is one class which uses TypedDict for input parameter groups (https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/modules/transformer/text_generation.py#L26). Everything else uses dataclass.

Reply:
Here's my concern. The hydra "_target_" pattern is typically a signal to the user that something could be customised using their own class. Here, this dataclass is intricately tied to a specific Dataset, making the `_target_` just a boilerplate in the log. Which is also easy to have a typo in.

Maybe the data class could be used only internally? Keeps the validation advantage.
```
class VocoderDataset(Dataset):
    def __init__(șelf, ..., dataset_meta: Dict[str, Dict], ...):
        ...
        for dataset_name, dataset in dataset_meta.items():
            dataset = DatasetMeta(**dataset)
            ...
```

Reply:
Makes sense to me. I will try to refactor it using the data class internally.

Reply:
Changed